TEST: added visual testing system

m-agour opened the pull request:
Added Snapshots analysis system.

Does this mean everything tested is working correctly? No, tests does not cover everything yet, this is just a starting point. 
How to proceed from here? whenever we see a bug we apply the visual test and see if the test can't detect the bug we can change the functionality of the snapshot tester and/or add more methods for visual analysis. So, we need to see if everything is workign as expected and before we fix we make this testing syste able to detect it first to prevent future bugs.

m-agour committed:
TEST: added visual testing system

skoudoro commented:
Are @maharshi-gor and you working on the same thing without communication ?

see #1050

I am lost

m-agour commented:
@skoudoro you are right Serge I didn't see maharshi's PR. they are almost the same. but here I try to use it to detect the errors we already have in fury and we test shading here which is crucial for testing the shaders parts. @maharshi-gor lets talk about this tomorrow.

m-agour committed:
FIX: Replace deprecated distutils.version with packaging.version

- Replaced LooseVersion from distutils.version with Version from packaging.version
- Fixes Python 3.12 compatibility where distutils was removed
- This was causing CI failures across all build matrix combinations

m-agour committed:
FIX: Correct doctest and test errors

- Fixed import in fury.testing doctest: use 'from fury.testing import' instead of 'from fury.testing.snapshot import'
- Fixed test_text_actor_translation_bug: changed 'color' parameter to 'colors' for fury_actor.text()
- Fixed fury.window.analyze_snapshot: changed 'background_threshold' parameter to 'bg_color' to match fury.testing.analyze_snapshot signature
- Fixed fury.window.snapshot doctest: added sphere actor to avoid 'no bounding sphere' error

skoudoro commented:
Did you and @maharshi-gor  talked ? 

What is the final conclusion ?
